packages/medusa/src/repositories/discount-condition.ts

```diff
@@ -34,61 +36,67 @@ type DiscountConditionResourceType = EntityTarget<
 
 @EntityRepository(DiscountCondition)
 export class DiscountConditionRepository extends Repository<DiscountCondition> {
-  getResourceIdentifiers(type: string): {
-    fromTable: DiscountConditionResourceType
-    resourceId: string | undefined
+  getJoinTableResourceIdentifiers(type: string): {
+    joinTableName: DiscountConditionResourceType
+    joinTableForeignKey: DiscountConditionJoinTableForeignKey | undefined
   } {
-    let fromTable: DiscountConditionResourceType = null
-    let resourceId: DiscountConditionResourceType | undefined = undefined
+    let joinTableName: DiscountConditionResourceType = null
+    let joinTableForeignKey: DiscountConditionJoinTableForeignKey | undefined =
+      undefined
 
     switch (type) {
       case DiscountConditionType.PRODUCTS: {
-        fromTable = DiscountConditionProduct
-        resourceId = DiscountConditionResourceTableId.PRODUCT_ID
+        joinTableName = DiscountConditionProduct
+        joinTableForeignKey = DiscountConditionJoinTableForeignKey.PRODUCT_ID
         break
       }
       case DiscountConditionType.PRODUCT_TYPES: {
-        fromTable = DiscountConditionProductType
-        resourceId = DiscountConditionResourceTableId.PRODUCT_TYPE_ID
+        joinTableName = DiscountConditionProductType
+        joinTableForeignKey =
+          DiscountConditionJoinTableForeignKey.PRODUCT_TYPE_ID
         break
       }
       case DiscountConditionType.PRODUCT_COLLECTIONS: {
-        fromTable = DiscountConditionProductCollection
-        resourceId = DiscountConditionResourceTableId.PRODUCT_COLLECTION_ID
+        joinTableName = DiscountConditionProductCollection
+        joinTableForeignKey =
+          DiscountConditionJoinTableForeignKey.PRODUCT_COLLECTION_ID
         break
       }
       case DiscountConditionType.PRODUCT_TAGS: {
-        fromTable = DiscountConditionProductTag
-        resourceId = DiscountConditionResourceTableId.PRODUCT_TAG_ID
+        joinTableName = DiscountConditionProductTag
+        joinTableForeignKey =
+          DiscountConditionJoinTableForeignKey.PRODUCT_TAG_ID
         break
       }
       case DiscountConditionType.CUSTOMER_GROUPS: {
-        fromTable = DiscountConditionCustomerGroup
-        resourceId = DiscountConditionResourceTableId.CUSTOMER_GROUP_ID
+        joinTableName = DiscountConditionCustomerGroup
+        joinTableForeignKey =
+          DiscountConditionJoinTableForeignKey.CUSTOMER_GROUP_ID
         break
       }
       default:
         break
     }
 
-    return { fromTable, resourceId }
+    return { joinTableName, joinTableForeignKey }
```

Comment:
What do you think of this naming? @zakariaelas 😁

Reply:
Looks great 👍 !